Tests/TensorFlowTests/LayerTests.swift

```diff
@@ -1647,8 +1647,37 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(transformerTensor.shape, transformerResult.shape)
     }
 
+    func testArray(){
+        var layers: [Dense<Float>] = []
+        let sizes = [(8, 7),(7, 6), (6, 5)]
+        for (inputSize, outputSize) in sizes {
+            let weight = Tensor<Float>(shape: [inputSize, outputSize], scalars: (0..<inputSize*outputSize).map(Float.init))
```

Comment:
```suggestion
            let weight = Tensor<Float>(shape: [inputSize, outputSize], scalars: (0..<Float(inputSize*outputSize)))
```

Does this work? If so, seems clearer.

Reply:
I don't think it works actually.

The current code maps a `Range<Int>` to `[Float]`, which is accepted by `Tensor.init(shape:scalars)`.

The new code tries to construct a `Range<Float>`, which works but cannot be passed to `Tensor.init(shape:scalars)` because it doesn't fulfill generic requirements. (`Range<Float>` is a "continuous" range, not a discrete one.)

Reply:
Yaa It is not working. It is throwing an error 
`error: initializer 'init(shape:scalars:on:)' requires that 'Float' conform to 'SignedInteger' let weight = Tensor<Float>(shape: [inputSize, outputSize], scalars: (0..<Float(inputSize*outputSize)))`

Reply:
Yeah, okay. Thanks for trying. Could we factor out a helper method for creating a Float tensor of consecutive scalars of a given size? This pattern is used a lot in this test, and it might be easier to read if there was a helper method instead.

Reply:
Are you saying different function to create tensor of specific size consisting of consecutive scalars?

Reply:
Yeah, I'm thinking something like the following:

```swift
func makeConsecutiveTensor(x: Int, y: Int) -> Tensor<Float> {
  // ...
}
```

Note: suggestions for alternate names for this function are welcome!

Reply:
It would be better if for creating consecutive integer we provide range function similar to [numpy.arange](https://docs.scipy.org/doc/numpy/reference/generated/numpy.arange.html) and [tf.range](https://www.tensorflow.org/api_docs/python/tf/range) and then reshape the output tensor as per our need.

```swift
func arange<Scalar: TensorFlowFloatingPoint> (_ x: Int, _ y: Int) -> Tensor<Scalar> {
  return Tensor<Scalar>(shape: [1, y-x], scalars: (x..<y).map(Scalar.init))
}
```
```swift
let x: Tensor<Float> = arange(2,8)
output: [[2.0, 3.0, 4.0, 5.0, 6.0, 7.0]]

x.reshaped(to:[3,2])
output: 
[[2.0, 3.0],
 [4.0, 5.0],
 [6.0, 7.0]]
```
It will also take care of data type of output variable that is needed

Reply:
> Could we factor out a helper method for creating a Float tensor of consecutive scalars of a given size? This pattern is used a lot in this test, and it might be easier to read if there was a helper method instead.

I think this `Tensor` initializer (using consecutive scalars) could be generally useful. If we are to write it, I think we should design something solid rather than adding a quick test utility.

I think we should consider this `Tensor` initializer after this PR, for separation of concerns! The current test code is fine.